### DISCUSSION TITLE
Buttons are now transparent since upgrade to V3

### DISCUSSION THREAD

**Discussion by thewhaleiam** [filamentphp/filament#9921] [Help]
Hi, just upgraded to filaments v3, all seems to have went well apart from buttons are not transparent do not apply the class colors. In the example screen shot the `Reject Content` and `Publish Content` are buttons and were working fine in V2.
<img width="1572" alt="Screenshot 2023-11-28 at 09 56 54" src="https://github.com/filamentphp/filament/assets/139074731/8d35a951-5a67-4a2a-8d33-19a9cf5da84b">

**Comment by leandrocfe** [comment]
Do you have a custom theme/css?

**Reply by thewhaleiam** [reply]
I have filament.css file which is pretty sparse. 

```
@import '../../vendor/filament/filament/resources/css/theme.css';

header.filament-sidebar-header {
    background-color: #1923ff;
}

mux-player::part(video) {
    height: 60vh;
}
```
 and mix file requires tailwindcss

const mix = require('laravel-mix');

```
/*
 |--------------------------------------------------------------------------
 | Mix Asset Management
 |--------------------------------------------------------------------------
 |
 | Mix provides a clean, fluent API for defining some Webpack build steps
 | for your Laravel applications. By default, we are compiling the CSS
 | file for the application as well as bundling up all the JS files.
 |
 */

mix.js('resources/js/app.js', 'public/js')
    .postCss('resources/css/app.css', 'public/css', [
        //
    ]);

mix.postCss('resources/css/filament.css', 'public/css', [
    require('tailwindcss'),
])
```

**Reply by leandrocfe** [reply]
Did you import via AppServiceProvider, right? Could you remove it to test?

**Reply by thewhaleiam** [reply]
Yes, it's loaded through a provider, when it is removed the styling is being applied to the buttons. What is the best approach to load my own styling with filament. This is how it is currently done.

```
    public function boot()
    {
        Filament::serving(function () {
            $this->registerPolicies();
            $this->registerNavigation();

            Filament::registerTheme(
                mix('css/filament.css')
            );
        });
    }
```

**Reply by leandrocfe** [reply]
A bit different on v3
https://filamentphp.com/docs/3.x/panels/themes#creating-a-custom-theme

**Reply by ahamilton9** [reply]
I'm in this same situation. I am building my theme via Mix with the postCss/tailwind plugin chain (I couldn't get the cli make method to give instructions for webpack or mix, so I had to wing it). I am including the theme in the panel provider with `->theme(asset('css/filament/admin/theme.css'))` which seems to include the file properly, but I still see transparent buttons and other UI elements. This may have something to do with the Mix build pipeline, but I'm unsure how a lot of this works under the hood. 

This discussion may also be related: https://github.com/filamentphp/filament/discussions/10161

**Reply by thewhaleiam** [reply]
I had to migrate my application from using mix to Vite. This [article](https://balajidharma.medium.com/how-to-migrating-from-laravel-mix-to-vite-fbb912237844) helped to make that change.

**Reply by ahamilton9** [reply]
I don't believe Vite is a requirement, and that would add a lot of time for our project. I will probably make an actual issue for this as it should be fixed or documented out.

**Comment by ahamilton9** [comment]
This appears to be some sort of bug with using Webpack/Laravel Mix to build the custom theme in Filament v3. However, I had a lot of trouble even getting Mix to build a new install of Filament _at all_. So what I've done is split our project to continue using Mix for most things and to use Vite solely for the custom theme as of right now. In case this helps someone in the future, here is my `vite.config.js` with plugins included for installation reference:
```js
import { defineConfig } from 'vite';
import laravel from 'laravel-vite-plugin';
import tailwindcss from '@vituum/vite-plugin-tailwindcss'

export default defineConfig({
    plugins: [
        laravel([
            'resources/css/filament/admin/theme.css'
        ]),
        tailwindcss(),
    ],
});
```
